lib/northstar.js

```
@@ -113,5 +113,7 @@ module.exports.fetchUserByMobile = function (mobile) {
  * @return {Promise}
  */
 module.exports.updateUser = function (userId, data) {
-  return exports.getClient().Users.update(userId, data);
+  return exports.getClient().Users.update(userId, data)
+    .then(res => res)
+    .catch(err => err);
```

This is no bueno. It creates a false positive response up the Promise chain. Either don't catch the error or reject.

---

Gotcha. This "fixed" receiving the Gambit reply, because it ignored the error we were getting for sending the Northstar update. I've updated the middleware to search for the specific Northstar error (and also fix the invalid `logger` call), but it'd be ideal to add this to all Northstar requests so don't get the "Internal Server Error" message back.